### PR TITLE
Add conversion to properly parse query parameter propagationPolicy

### LINKED ISSUE #43329: API server cannot parse query parameter propagationPolicy sent with a delete request
https://github.com/kubernetes/kubernetes/pull/35806 makes API server accepts DeleteOptions as a query parameter, but APIserver cannot parse `?propagationPolicy=orphan`, which is a new field added in 1.6.
```bash
s$ curl -X DELETE http://localhost:8080/api/v1/namespaces/default/replicationcontrollers/redis?propagationPolicy=Orphan
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "converting (url.Values).[]string.[]string to (v1.DeleteOptions).*v1.DeletionPropagation.v1.DeletionPropagation: couldn't copy '[]string' into 'v1.DeletionPropagation'; didn't understand types",
  "code": 500
}
```
Probably it's a bug in the parameterCodec.

cc @lavalamp @smarterclayton @bdbauer

### PR DESCRIPTION
**Pull request by roycaihw**
**What this PR does / why we need it**:
This delete request query parameter has been broken for a year. I'm not sure if we want to deprecate the non-standard DeleteOptions body on delete request eventually. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #43329

**Special notes for your reviewer**:

**Release note**:

```release-note
API server properly parses propagationPolicy as a query parameter sent with a delete request
```

/sig api-machinery

### CODE REVIEW

**Comment by liggitt** [comment]
can you add a test for this?

**Comment by roycaihw** [comment]
@liggitt do we want a unit test for the conversion function, or do we have some place to test the behavior of api server delete handler?

**Comment by roycaihw** [comment]
unit test for the conversion function added

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go [hunk 1]
```diff
@@ -304,3 +306,13 @@ func Convert_Slice_string_To_Slice_int32(in *[]string, out *[]int32, s conversio
 	}
 	return nil
 }
+
+// Convert_Slice_string_To_v1_DeletionPropagation allows converting a URL query parameter propagationPolicy
+func Convert_Slice_string_To_v1_DeletionPropagation(input *[]string, out *DeletionPropagation, s conversion.Scope) error {
+	if len(*input) > 0 {
```
Comment (ncdc) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go]: Don't you need to check if `*input` is `nil` first?

Comment (roycaihw) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go]: We don't check if `*input` and `*output` are `nil` in conversion functions, e.g. https://github.com/kubernetes/kubernetes/blob/7e75a09db6dc13218b4f5e06493115ae96f83774/staging/src/k8s.io/apimachinery/pkg/runtime/conversion.go
https://github.com/kubernetes/kubernetes/blob/7e75a09db6dc13218b4f5e06493115ae96f83774/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1beta1/conversion.go
I will take another look to see where is it enforced.

Comment (roycaihw) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go]: It's enforced by 
https://github.com/kubernetes/kubernetes/blob/bed45311315f63f7b0ddce756559920c0139273e/staging/src/k8s.io/apimachinery/pkg/conversion/converter.go#L497
through
https://github.com/kubernetes/kubernetes/blob/bed45311315f63f7b0ddce756559920c0139273e/staging/src/k8s.io/apimachinery/pkg/conversion/helper.go#L27-L39

Comment (ncdc) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go]: Ok thanks 👍

**Comment by roycaihw** [comment]
@liggitt Would like another round of review :)

**Pull request merged**